corehq/apps/accounting/payment_handlers.py

```diff
@@ -394,3 +378,65 @@ def get_email_context(self):
             'items': self._humanized_products() + self._humanized_features()
         })
         return context
+
+
+class AutoPayInvoicePaymentHandler(object):
+    def pay_autopayable_invoices(self, date_due):
+        """ Pays the full balance of all autopayable invoices on date_due """
+        autopayable_invoices = Invoice.autopayable_invoices(date_due)
+        for invoice in autopayable_invoices:
+            logging.info("[Billing][Autopay] Autopaying invoice {}".format(invoice.id))
+            amount = invoice.balance.quantize(Decimal(10) ** -2)
+
+            auto_payer = invoice.subscription.account.auto_pay_user
+            payment_method = StripePaymentMethod.objects.get(web_user=auto_payer)
+            autopay_card = payment_method.get_autopay_card(invoice.subscription.account)
+            if autopay_card is None:
+                continue
+
+            try:
+                payment_record = payment_method.create_charge(autopay_card, amount_in_dollars=amount)
+            except stripe.error.CardError:
+                self._handle_card_declined(invoice, payment_method)
+                continue
+            except payment_method.STRIPE_GENERIC_ERROR as e:
+                self._handle_card_errors(invoice, payment_method, e)
+                continue
+            else:
+                invoice.pay_invoice(payment_record)
+                self._send_payment_receipt(invoice, payment_record)
+
+    def _send_payment_receipt(self, invoice, payment_record):
+        from corehq.apps.accounting.tasks import send_purchase_receipt
+        try:
```

Comment:
nit: seems like a few these lines could be taken outside of the try catch. mainly the variable declarations